contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -296,6 +298,32 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		}
 
 		witnessUpdates = preimageSubscription.WitnessUpdates
+
+		// Notify the htlc interceptor. There may be a client connected
```

Comment:
Here's what I mean joost: we'd take this logic and pass in the full HTLC and payload information into `SubscribeUpdates`. Then all the other logic here stays as normal, the the contract court doesn't even need to know about the interceptable switch stuff. Along the way, an import cycle wouldn't be introduced, meaning we don't have to make that new package at all.

Reply:
Okay, you want to take it another step further. Made the change.